### ISSUE TITLE
Electron won't start from terminal

### ISSUE THREAD

**Issue by biomade** [bus-stop/Termination#35]
I was beating on this for hours... npm start will not launch an electron application and gives a long debug list. but when I open a regular command prompt and run npm start it does!
I am using windows 10 and Atom.
Here is the debug log 

0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'start' ]
2 info using npm@3.10.10
3 info using node@v6.10.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle electronstart@1.0.0~prestart: electronstart@1.0.0
6 silly lifecycle electronstart@1.0.0~prestart: no script for prestart, continuing
7 info lifecycle electronstart@1.0.0~start: electronstart@1.0.0
8 verbose lifecycle electronstart@1.0.0~start: unsafe-perm in lifecycle true
9 verbose lifecycle electronstart@1.0.0~start: PATH: C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;C:\Users\laurie\Documents\GitHub\ElectronStart\node_modules\.bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\ManagementStudio\;C:\Users\laurie\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files (x86)\GtkSharp\2.12\bin;C:\Program Files\Git\cmd;C:\Program Files\nodejs\;C:\Users\laurie\AppData\Local\Microsoft\WindowsApps;C:\Users\laurie\AppData\Local\atom\bin;C:\Users\laurie\AppData\Roaming\npm
10 verbose lifecycle electronstart@1.0.0~start: CWD: C:\Users\laurie\Documents\GitHub\ElectronStart
11 silly lifecycle electronstart@1.0.0~start: Args: [ '/d /s /c', 'electron main.js' ]
12 silly lifecycle electronstart@1.0.0~start: Returned: code: 1  signal: null
13 info lifecycle electronstart@1.0.0~start: Failed to exec start script
14 verbose stack Error: electronstart@1.0.0 start: `electron main.js`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:255:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:877:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid electronstart@1.0.0
16 verbose cwd C:\Users\laurie\Documents\GitHub\ElectronStart
17 error Windows_NT 10.0.14393
18 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
19 error node v6.10.0
20 error npm  v3.10.10
21 error code ELIFECYCLE
22 error electronstart@1.0.0 start: `electron main.js`
22 error Exit status 1
23 error Failed at the electronstart@1.0.0 start script 'electron main.js'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the electronstart package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     electron main.js
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs electronstart
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls electronstart
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by Fred-Barclay** [comment]
G'day @biomade and thanks for trying Termination!
Sorry about that - it's definitely a bug. We just fixed this for Linux (and macOS, I believe), but Windows is still giving us fits! :unamused: @PhillMcKraken has done some much-appreciated digging and it appears the issue is with the pty.js builds for Windows that we rely on. 

I'm currently exploring ways around this.

**Comment by biomade** [comment]
Glad to hear that you are aware of it. Thanks for the reply and for the time being, now I know what to do to keep moving forward. I look forward to the update

**Comment by PhillMcKraken** [comment]
I have confirmed moving to node-pty fixes this, however we loose prebuilt binaries.

**Comment by Fred-Barclay** [comment]
@PhillMcKraken Can you walk me through replacing pty.js with node-pty?
I've checked https://github.com/PhillMcKraken/platformio-atom-ide-terminal/commit/5653f119d225f09cd6053d5b9ea1b267542a228c and I see the two lines that you've edited, but after editing those lines in my local installed Termination folder (.atom/packages/termination) and running `Update Package Dependencies: Update` inside Atom, all I get is a terminal window with a flashing cursor.
What's the proper way to go about this?

**Comment by PhillMcKraken** [comment]
Whats your console say? What I did was I ran "apm reinstall" inside the package from my terminal and that fixed it all.

**Comment by Fred-Barclay** [comment]
The console reports it can't find `node-pty`. 
Let me try `apm reinstall`. Back in a minute or two...

**Comment by PhillMcKraken** [comment]
just to be sure, you did make sure note-pty installed inside the termination/node_modules, right?

**Comment by PhillMcKraken** [comment]
also i probably goofed, you may want to point right at "node-pty" in package.json (I manually ran npm install for it)

**Comment by Fred-Barclay** [comment]
Yep, it was definitely installed in node_modules. 
Can you specify more about pointing right at "node-pty" in package.json? This is what mine looks like currently:
``` "dependencies": {
blah
    "node-pty": "https://github.com/Tyriar/node-pty-tarball/master",
```

**Comment by PhillMcKraken** [comment]
have it install from npmjs like a normal dependency

`"node-pty": "^0.6.2"`

**Comment by Fred-Barclay** [comment]
```
apm reinstall
Unrecognized command: reinstall
```
apm rebuild works though. 

Still got a flashing cursor... I'll try editing package.json like your last post.

**Comment by PhillMcKraken** [comment]
yeah its apm rebuild.

To do exactly what I did when I confirmed it manually, I was at termination in my terminal, ran `npm install node-pty` then `apm rebuild`, updated process.coffee to point at node-pty instead, and then it worked just fine. My main issues were before I knew about `apm rebuild` so I was getting invalid native module errors due to the version mismatches.

**Comment by Fred-Barclay** [comment]
Bleh - still same flashing cursor.
I'm going to have to call it quits for now. Back at it tomorrow!

And thanks for all your help! :+1: Windows is a weak point with me (okay, if it's not Linux it's a weak point :grin:) so I really appreciate having someone who knows more about the platform around and digging - makes this ten times easier!

**Comment by PhillMcKraken** [comment]
strange, no idea why it wouldnt find a module inside node_modules -- has to be a path issue of some sort...

`.atom\packages\platformio-ide-terminal\node_modules\node-pty`
`.atom\packages\platformio-ide-terminal\lib\process.coffee` 

Why wouldnt it locate it? Odd.

**Comment by the-j0k3r** [comment]
this has been fixed by node-pty-prebuilt and atom 1.41.0 and newer
